### PR TITLE
x11-libs/libxkbcommon: Add support for wayland

### PR DESCRIPTION
**Pull request by lanodan**
Signed-off-by: Haelwenn (lanodan) Monnier <contact@hacktivis.me>

### CODE REVIEW

File: x11-libs/libxkbcommon/libxkbcommon-1.3.0.ebuild [hunk 1]
```diff
@@ -18,17 +18,22 @@ inherit meson multilib-minimal ${GIT_ECLASS} python-any-r1 virtualx
...
 RESTRICT="!test? ( test )"
 SLOT="0"
 
 BDEPEND="
 	sys-devel/bison
 	doc? ( app-doc/doxygen )
 	test? ( ${PYTHON_DEPS} )
+	wayland? (
+		dev-util/wayland-scanner
+		>=dev-libs/wayland-protocols-1.12
```
Comment (mattst88) [inline x11-libs/libxkbcommon/libxkbcommon-1.3.0.ebuild:31]: I think `wayland-protocols` should be in `DEPEND`.

File: x11-libs/libxkbcommon/libxkbcommon-1.3.0.ebuild [hunk 2]
```diff
@@ -45,7 +50,7 @@ multilib_src_configure() {
 	local emesonargs=(
 		-Ddefault_library="$(usex static-libs both shared)"
 		-Dxkb-config-root="${EPREFIX}/usr/share/X11/xkb"
-		-Denable-wayland=false # Demo applications
+		$(meson_use wayland enable-wayland)
```
Comment (mattst88) [inline x11-libs/libxkbcommon/libxkbcommon-1.3.0.ebuild:53]: This controls the building and installation of
````
./usr/libexec/xkbcommon/xkbcli-interactive-wayland
./usr/share/man/man1/xkbcli-interactive-wayland.1.bz2
````

I see that we're already installing

````
./usr/libexec/xkbcommon/xkbcli-interactive-x11
./usr/share/man/man1/xkbcli-interactive-x11.1.bz2
````

with USE=X. (It also controls some other things)

So... seems fine. Is this tool useful for you? I'm just unsure of the rationale.

**Review by mattst88 — CHANGES_REQUESTED** [review]

File: x11-libs/libxkbcommon/libxkbcommon-1.3.0.ebuild (hunk 2 above)
Comment (lanodan) [reply x11-libs/libxkbcommon/libxkbcommon-1.3.0.ebuild:53]: Yeah, the USE=X requires to have `xorg-server` installed, which isn't required for wayland usage in gentoo and I'm not aware of another wayland tool similar to this one or x11-apps/xev (`WAYLAND_DEBUG=1` is nice but it doesn't decodes events in something a human could directly use).

**Comment by mattst88** [comment]
Thanks!

**Pull request closed without merging**